addons/longhorn/1.1.1/install.sh

```diff
@@ -119,3 +122,38 @@ function longhorn_preflight() {
     local src="${DIR}/addons/longhorn/${LONGHORN_VERSION}"
     echo "${src}/host-preflight.yaml"
 }
+
+function check_mount_propagation() {
+    local src=$1
+
+    kubectl get ns longhorn-system >/dev/null 2>&1 || kubectl create ns longhorn-system >/dev/null 2>&1
+    kubectl delete -n longhorn-system ds longhorn-environment-check || true
+
+    render_yaml_file "$src/tmpl-mount-propagation.yaml" > "$src/mount-propagation.yaml"
+    kubectl apply -f "$src/mount-propagation.yaml"
+    echo "Waiting for the Longhorn Mount Propagation Check Daemonset to be ready"
+    spinner_until 120 longhorn_daemonset_is_ready longhorn-manager
+
+    validate_longhorn_ds
+
+    kubectl delete -f "$src/mount-propagation.yaml"
+}
+
+function validate_longhorn_ds() {
+    local allSupported=true
+
+    local allpods=$(kubectl get daemonsets -n longhorn-system longhorn-environment-check --no-headers | tr -s ' ' | cut -d ' ' -f4)
+    local bidirectional=$(kubectl get pods -l app=longhorn-environment-check -o=jsonpath='{.items[*].spec.containers[0].volumeMounts[*]}' | grep -o 'Bidirectional' | wc -l)
+
+    if [ "$allpods" == "" ]; then
+        logWarn "unable to determine health and status of longhorn-environment-check daemonset"
+    else
+        if [ "$bidirectional" -lt "$allpods" ]; then
+            logWarn "Only $bidirectional of $allpods nodes support Longhorn storage"
```

Comment:
I'm unclear the point of this check. In the linked preflight from the upstream this would block the installation.

Reply:
I'll change it to exit with an error if no nodes work

Reply:
I've added a 'bail if none are detected' condition

But our other preflights continue if there's any chance of a successful install

Reply:
i still just dont think i understand the goal here. are we trying to match the logic in the longhorn preflight? if so this doesnt seem to match

```
  if [ "$allSupported" != "true" ]; then
   ...
   exit 1
```